src/pytorch_lightning/core/datamodule.py

```diff
@@ -180,8 +180,8 @@ def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
     @classmethod
     def load_from_checkpoint(
         cls,
-        checkpoint_path: Union[_PATH, IO],
-        hparams_file: Optional[_PATH] = None,
+        checkpoint_path: Union[str, Path],
+        hparams_file: Optional[Union[str, Path]] = None,
```

Comment:
Why are we making these changes?
Also, can you show why IO is not supported?

Reply:
I removed IO because I understood it to be generic, similar to using Any. Also, while reading typing docs, I found that IO is deprecated and will be removed in a future release of Python. https://docs.python.org/3/library/typing.html#typing.IO

Reply:
reverted to https://github.com/Lightning-AI/lightning/pull/13932/commits/0502a4e9fe2f1b229265de8ae82a129e704f65ee and synced with upstream in https://github.com/Lightning-AI/lightning/pull/13932/commits/82fc4528431fd3412e30782ba64bcc65d2adba38